pkg/apis/core/validation/validation.go

```diff
@@ -1667,12 +1701,15 @@ func ValidatePersistentVolumeSpec(pvSpec *core.PersistentVolumeSpec, pvName stri
 		allErrs = append(allErrs, field.Required(fldPath.Child("accessModes"), ""))
 	}
 	for _, mode := range pvSpec.AccessModes {
+		if opts.AllowReadWriteOncePod && mode == core.ReadWriteOncePod {
```

Comment:
nit: prefer to use the option to add the value to `supportedAccessModes` (which should be a local variable).

Reply:
Done

Reply:
you should insert the element once before the loop, regardless of the usage. All that matters is AllowReadWriteOncePod.